### PR TITLE
HPCC-31715 Change the default field translation mode

### PR DESCRIPTION
**Pull request by jackdelv**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jackdelv** [comment]
@ghalliday I have added the code for processing the fields in this commit alongside the change to the default for the tests to pass. Let me know if you wanted just the changes to the default.

**Comment by ghalliday** [comment]
@jackdelv please can you rebase onto the lastest master - it will remove some of these changes.

**Comment by jackdelv** [comment]
@ghalliday Rebased to latest master.

File: initfiles/componentfiles/configschema/xsd/eclagent.xsd [hunk 1]
```diff
@@ -57,13 +57,14 @@
...
-                                  hpcc:presetValue="payload"
+                                  hpcc:presetValue="payloadRemoveOnly"
                                   hpcc:tooltip="Enables translation (where possible) of mismatched index layouts on-the-fly. Specify 'payload' to attempt to translate payload fields only">
                         <xs:simpleType>
                             <xs:restriction base="xs:string">
                                 <xs:enumeration value="false" hpcc:displayName="False" hpcc:desciption=""/>
-                                <xs:enumeration value="true" hpcc:displayName="true" hpcc:description=""/>
+                                <xs:enumeration value="true" hpcc:displayName="True" hpcc:description=""/>
                                 <xs:enumeration value="payload" hpcc:displayName="Payload" hpcc:description=""/>
+                                <xs:enumeration value="payloadRemoveOnly" hpcc:displayName="PayloadRemoveOnly" hpcc:description=""/>
```
Comment (ghalliday) [inline initfiles/componentfiles/configschema/xsd/eclagent.xsd:67]: This looks like a change that should be in 9.4.x.  I will tag the others as well.

File: initfiles/componentfiles/configschema/xsd/roxie.xsd [hunk 2]
```diff
@@ -110,13 +110,14 @@
...
-                                  hpcc:presetValue="payload"
+                                  hpcc:presetValue="payloadRemoveOnly"
                                   hpcc:tooltip="Enables translation (where possible) of mismatched index layouts on-the-fly. Specify 'payload' to attempt to translate payload fields only">
                         <xs:simpleType>
                             <xs:restriction base="xs:string">
                                 <xs:enumeration value="false" hpcc:displayName="False" hpcc:desciption=""/>
-                                <xs:enumeration value="true" hpcc:displayName="true" hpcc:description=""/>
+                                <xs:enumeration value="true" hpcc:displayName="True" hpcc:description=""/>
                                 <xs:enumeration value="payload" hpcc:displayName="Payload" hpcc:description=""/>
+                                <xs:enumeration value="payloadRemoveOnly" hpcc:displayName="PayloadRemoveOnly" hpcc:description=""/>
```
Comment (ghalliday) [inline initfiles/componentfiles/configschema/xsd/roxie.xsd:120]: This looks like a change that should be in 9.4.x

File: initfiles/componentfiles/configschema/xsd/thor.xsd [hunk 3]
```diff
@@ -224,13 +224,14 @@
...
-                              hpcc:presetValue="payload"
+                              hpcc:presetValue="payloadRemoveOnly"
                               hpcc:tooltip="Enables translation (where possible) of mismatched index layouts on-the-fly. Specify 'payload' to attempt to translate payload fields only">
                     <xs:simpleType>
                         <xs:restriction base="xs:string">
                             <xs:enumeration value="false" hpcc:displayName="False" hpcc:desciption=""/>
-                            <xs:enumeration value="true" hpcc:displayName="true" hpcc:description=""/>
+                            <xs:enumeration value="true" hpcc:displayName="True" hpcc:description=""/>
                             <xs:enumeration value="payload" hpcc:displayName="Payload" hpcc:description=""/>
+                            <xs:enumeration value="payloadRemoveOnly" hpcc:displayName="PayloadRemoveOnly" hpcc:description=""/>
```
Comment (ghalliday) [inline initfiles/componentfiles/configschema/xsd/thor.xsd:234]: This looks like a change that should be in 9.4.x

File: initfiles/componentfiles/configxml/eclagent_config.xsd.in [hunk 4]
```diff
@@ -201,6 +201,7 @@
           <xs:enumeration value="false"/>
           <xs:enumeration value="true"/>
           <xs:enumeration value="payload"/>
+          <xs:enumeration value="payloadRemoveOnly"/>
```
Comment (ghalliday) [inline initfiles/componentfiles/configxml/eclagent_config.xsd.in:204]: This looks like a change that should be in 9.4.x

File: initfiles/componentfiles/configxml/roxie.xsd.in [hunk 5]
```diff
@@ -649,6 +649,7 @@
           <xs:enumeration value="false"/>
           <xs:enumeration value="true"/>
           <xs:enumeration value="payload"/>
+          <xs:enumeration value="payloadRemoveOnly"/>
```
Comment (ghalliday) [inline initfiles/componentfiles/configxml/roxie.xsd.in:652]: This looks like a change that should be in 9.4.x

File: initfiles/componentfiles/configxml/thor.xsd.in [hunk 6]
```diff
@@ -688,6 +688,7 @@
           <xs:enumeration value="false"/>
           <xs:enumeration value="true"/>
           <xs:enumeration value="payload"/>
+          <xs:enumeration value="payloadRemoveOnly"/>
```
Comment (ghalliday) [inline initfiles/componentfiles/configxml/thor.xsd.in:691]: This looks like a change that should be in 9.4.x

File: testing/regress/ecl/layouttrans.ecl [hunk 7]
```diff
@@ -26,7 +26,7 @@ version := #IFDEFINED(root.version, 1);
 
 //--- end of version configuration ---
 
-#option ('layoutTranslation', true);
+#option ('layoutTranslation', 'payload');
```
Comment (ghalliday) [inline testing/regress/ecl/layouttrans.ecl:29]: This looks like a change that should be in 9.4.x

File: testing/regress/ecl/sqtrans.ecl [hunk 8]
```diff
@@ -22,6 +22,8 @@ multiPart := #IFDEFINED(root.multiPart, false);
 
 //--- end of version configuration ---
 
+#option('layoutTranslation', 'payload');
```
Comment (ghalliday) [inline testing/regress/ecl/sqtrans.ecl:25]: This looks like a change that should be in 9.4.x

File: testing/regress/ecl/translatedisk.ecl [hunk 9]
```diff
@@ -27,7 +27,7 @@ optRemoteRead := #IFDEFINED(root.optRemoteRead, true);
 
 #onwarning(2036, ignore);
 #onwarning(4522, ignore);
-#option ('layoutTranslation', true);
+#option ('layoutTranslation', 'payload');
```
Comment (ghalliday) [inline testing/regress/ecl/translatedisk.ecl:30]: This looks like a change that should be in 9.4.x

**Review by ghalliday — COMMENTED** [review]
@jackdelv a few changes that should be in 9.4 rather than this - which should only change the defaults.  If you agree please split the PR In two again.  Thanks.

**Comment by jackdelv** [comment]
@ghalliday I have rebased to remove the changes that went into 9.4.x.

File: helm/hpcc/values.schema.json [hunk 10]
```diff
@@ -1865,8 +1865,8 @@
           "description": "Should the index node memory allocation flush the cache and retry if memory allocation fails"
         },
         "fieldTranslationEnabled": {
-          "default": "payload",
-          "enum": ["false", "true", "payload"],
```
Comment (ghalliday) [inline helm/hpcc/values.schema.json:1869]: @jackdelv this change also needs to be in the previous build otherwise the option cannot be configured.

Comment (jackdelv) [reply helm/hpcc/values.schema.json:1869]: Opened #18694

**Comment by jackdelv** [comment]
Rebased to remove change from #18694

**Review by ghalliday — APPROVED** [review]

**Pull request merged**